staging/src/k8s.io/apiserver/pkg/endpoints/request/requestinfo.go

```diff
@@ -246,6 +251,22 @@ func (r *RequestInfoFactory) NewRequestInfo(req *http.Request) (*RequestInfo, er
 	return &requestInfo, nil
 }
 
+func hasNamespace(apiPrefix string, currentParts []string) bool {
+	if len(currentParts) <= 1 {
+		return false
+	}
+	if apiPrefix != "apis" {
+		return true
+	}
+	if len(currentParts) == 2 {
+		return false
+	}
+	if len(currentParts) == 3 && crSubresources.Has(currentParts[2]) {
+		return false
+	}
+	return true
+}
```

Comment:
Could you document what the flow of logic here is doing similar to how it's done above?

Reply:
I added some comments for the steps taken and also added a `NOTE` mentioning the issue about resources with CR subresource-named plural names (such as a resource with Kind "Statu", please see the description of this PR for a full example).